test/formats/lessIcons.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+var assert  = require('chai').assert,
+    formats = require('../../lib/common/formats');
+
+var file = {
+  "destination": "output/",
+  "format": "less/icons",
+  "name": "foo"
+};
+
+var propertyName = "content-icon-email";
+var propertyValue = "'\\E001'";
+var itemClass = "3d_rotation";
+
+var dictionary = {
+  "allProperties": [{
+    "name": propertyName,
+    "value": propertyValue,
+    "original": {
+      "value": propertyValue
+    },
+    "attributes": {
+      "category": "content",
+      "type": "icon",
+      "item": itemClass
+    }
+  }]
+};
+
+var config = {
+  prefix: 'sd' // Style-Dictionary Prefix
+};
+
+
+var formatter = formats['less/icons'].bind(file);
+
+describe('formats', function() {
+  describe('less/variables', function() {
+    var variableResult = '@' + propertyName;
+    it('should have a valid less variable', function() {
+      var expectedResult = variableResult + ': ' + propertyValue + ';';
+      // Check for substring of formatter function
+      assert.include(formatter(dictionary, config), expectedResult, 'string contains substring');
+    });
+
+    it('should have a valid class created', function() {
+      var expectedResult = '.' + config.prefix + '-icon.' + itemClass + ':before { content: ' + variableResult + '; }';
```

Comment:
kinda didn't like this approach to test which is based on substrings. Any good way to test this? I didn't find any examples for scss variables testing

Reply:
Agreed, this testing is a bit brittle. I'm open to ideas for how to improve this. We could have the less/sass compilers as dev dependencies and try compiling the output file?

Reply:
sounds like a good idea; Let me give it a shot